tests/unit/acquisition/test_rule.py

```diff
@@ -186,3 +189,37 @@ def test_trust_region_for_unsuccessful_local_to_global_trust_region_reduced() ->
     npt.assert_array_less(current_state.eps, previous_state.eps)  # current TR smaller than previous
     assert current_state.is_global
     npt.assert_array_almost_equal(current_state.acquisition_space.lower, lower_bound)
+
+
+class _BatchModelMinusMeanMaximumSingleBuilder(BatchAcquisitionFunctionBuilder):
+    def prepare_acquisition_function(
+        self, dataset: Mapping[str, Dataset], model: Mapping[str, ProbabilisticModel]
+    ) -> BatchAcquisitionFunction:
+        return lambda at: -tf.reduce_max(model[OBJECTIVE].predict(at)[0], axis=-2)
+
+
+def test_batch_acquisition_returns_batches_of_right_size() -> None:
+    search_space = Box(tf.constant([-2.2, -1.0]), tf.constant([1.3, 3.3]))
+    num_query_points = 3
+    ego = BatchAcquisitionRule(
+        num_query_points,
+        _BatchModelMinusMeanMaximumSingleBuilder())
+    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    query_point, _ = ego.acquire(
+        search_space, {OBJECTIVE: dataset}, {OBJECTIVE: QuadraticWithUnitVariance()}
+    )
+    assert query_point.shape == [num_query_points, 2]
+
+
+def test_batch_acquisition_finds_minimum() -> None:
+    search_space = Box(tf.constant([-2.2, -1.0]), tf.constant([1.3, 3.3]))
+    expected_minimum = tf.constant([0., 0.])
+    num_query_points = 4
+    ego = BatchAcquisitionRule(
+        num_query_points,
+        _BatchModelMinusMeanMaximumSingleBuilder())
+    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    query_point, _ = ego.acquire(
+        search_space, {OBJECTIVE: dataset}, {OBJECTIVE: QuadraticWithUnitVariance()}
+    )
+    npt.assert_allclose(query_point - expected_minimum[None, :], 0., atol=1e-3)
```

Comment:
if you don't use broadcasting in `assert_allclose`, you can merge these two tests into one
```suggestion
    npt.assert_allclose(query_point, [[0., 0.]] * num_query_points, atol=1e-3)
```